lib/schedule.js

```diff
@@ -23,9 +25,9 @@ Schedule.prototype.now = function() {
   return +new clock.Date();
 };
 
-Schedule.prototype.run = function(task, timeout) {
+Schedule.prototype.run = function(task, timeout, dontExecuteIfLate, rescheduleIfLate) {
```

Comment:
There are three cases when this can be called.
An alternative might be to pass in a `mode` arg instead of two booleans which might make this more readable.

Essentially the modes are:
1. `ack` - we always want to call this task `ASAP`,
2. `checkReclaim` - this we should `RESCHEDULE` until the host is under load, and
3. `tryReclaim` - this should be `ABANDON` ed if the host is under load